tinygrad/opsgpu.py

```diff
@@ -233,13 +233,20 @@ def backward(ctx, grad_output):
 
 class Sum(Function):
   @staticmethod
-  def forward(ctx, input):
-    ctx.save_for_backward(input)
-    return reduce_op(ctx, "out += a", "out", input)
+  def forward(ctx, input, axis=None):
+    ctx.save_for_backward(input, axis)
+    ret = reduce_op(ctx, "out += a", "out", input, axis=axis)
+    if axis is not None:
+        ret.shape = tuple([input.shape[i] for i in range(len(input.shape)) if i not in axis])
+    return ret
 
   @staticmethod
   def backward(ctx, grad_output):
-    input, = ctx.saved_tensors
+    input, axis = ctx.saved_tensors
+
+    output = GPUBuffer([1 if axis is None or i in axis else input.shape[i] for i in range(len(input.shape))], hostbuf=grad_output)
```

Comment:
This line is hard to read

Reply:
agree, was surprised that works! I adapt the np.array from reduce